### PR TITLE
UpdateName state

### PR DESCRIPTION
**Pull request by humulla**
## Link to JIRA ticket

[Jira Ticket](https://hackney.atlassian.net/browse/MTTL-3113)
## Describe this PR

Trigget last state of the CON process and update the name in person API and raise event

#### _Checklist_

- [x] Added end-to-end (i.e. integration) tests where necessary e.g to test complete functionality of a newly added feature
- [x] Added tests to cover all new production code
- [x] Checked all code for possible refactoring
- [x] Code pipeline builds correctly

### CODE REVIEW

File: ProcessesApi/V1/Helpers/SharedDbOperationsHelper.cs [hunk 1]
```diff
@@ -16,30 +16,40 @@
...
+using Hackney.Shared.Person.Factories;
+using Hackney.Shared.Person.Boundary.Request;
+using ProcessesApi.V1.Boundary.Request;
+using System.Text.Json;
+using System.Text.Json.Serialization;
 
 namespace ProcessesApi.V1.Helpers
 {
-    public class SoleToJointDbOperationsHelper : ISoleToJointDbOperationsHelper
+    public class SharedDbOperationsHelper : ISharedDbOperationsHelper
```
Comment (notatiyyah) [inline ProcessesApi/V1/Helpers/SharedDbOperationsHelper.cs:29]: Maybe just DbOperationsHelper?

**Review by rcmsoftwaredevelopment — APPROVED** [review]

File: ProcessesApi/V1/Helpers/SharedDbOperationsHelper.cs [hunk 2]
```diff
@@ -247,5 +257,50 @@ public async Task<Guid> UpdateTenures(Process process, Token token)
...
+                personRequestObject.Title = (Title?) nameSubmitted.GetValueOrDefault(ChangeOfNameKeys.Title);
+
+            var result = await _personDbGateway.UpdatePersonByIdAsync(process.TargetId, personRequestObject).ConfigureAwait(false);
+
+            var message = _personSnsFactory.Update(result, _token);
+            var topicArn = Environment.GetEnvironmentVariable("PERSON_SNS_ARN");
+            await _snsGateway.Publish(message, topicArn).ConfigureAwait(false);
+        }
+
+        public Task UpdatePerson(Process process, Token token, ProcessTrigger processTrigger)
```
Comment (notatiyyah) [inline ProcessesApi/V1/Helpers/SharedDbOperationsHelper.cs:299]: Is this used?

File: ProcessesApi/V1/Helpers/SharedDbOperationsHelper.cs [hunk 3]
```diff
@@ -247,5 +257,50 @@ public async Task<Guid> UpdateTenures(Process process, Token token)
         }
 
         #endregion
+
+        #region Update Name
+
+        protected JsonSerializerOptions CreateJsonOptions()
```
Comment (notatiyyah) [inline ProcessesApi/V1/Helpers/SharedDbOperationsHelper.cs:263]: Is this used?

File: ProcessesApi/V1/Helpers/SharedDbOperationsHelper.cs [hunk 4]
```diff
@@ -247,5 +257,50 @@ public async Task<Guid> UpdateTenures(Process process, Token token)
...
+                {
+                    (JsonConverter)new JsonStringEnumConverter()
+                }
+            };
+        }
+        public async Task UpdatePerson(Process process, Token token)
+        {
+            _token = token;
+
+            var existingPerson = await _personDbGateway.GetPersonById(process.TargetId).ConfigureAwait(false);
```
Comment (notatiyyah) [inline ProcessesApi/V1/Helpers/SharedDbOperationsHelper.cs:279]: I don't think this db call is necessary - the variable isn't used

**Review by notatiyyah — APPROVED** [review]

**Pull request merged**